packages/rmf-auth/lib/keycloak.ts

```diff
@@ -57,17 +57,35 @@ export default class KeycloakAuthenticator
       const refreshed = await this._inst.updateToken(30);
       refreshed && debug('token refreshed');
     } catch {
-      debug('token no refreshed');
+      debug('token not refreshed');
     }
 
-    this._user = this._inst.idTokenParsed && {
+    this._user = this._inst.tokenParsed && {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       username: (this._inst.idTokenParsed as any).preferred_username,
-      token: this._inst.idToken || '',
+      token: this._inst.token || '',
     };
     this._initialized = true;
   }
 
+  async refreshToken(): Promise<void> {
+    // check and update the token 5 seconds prior to expiry
+    if (this._initialized) {
+      const refreshed = await this._inst.updateToken(5);
+      if (refreshed) {
+        this._user = {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          username: (this._inst.idTokenParsed as any).preferred_username,
+          token: this._inst.token || '',
+        };
+        this.emit('tokenRefresh', null);
+      } else {
+        debug('token not refreshed');
```

Comment:
throw an error?

Reply:
Is this the same as above?

Reply:
I think it is not good to throw an error here. If the token is not ready for refreshing because it is not `min validity` seconds prior to expiry, the error will break the whole system when it is functioning perfectly well.

Reply:
yeah